packages/net/http/_test.pony

```diff
@@ -362,3 +368,212 @@ primitive _Test
     h.assert_eq[String](path, url.path)
     h.assert_eq[String](query, url.query)
     h.assert_eq[String](fragment, url.fragment)
+
+// Actor and classes to test the HTTPClient and modified _HTTPConnection.
+class _HTTPConnTestHandler is HTTPHandler
+  var n_received: U32 = 0
+  let h: TestHelper
+
+  new create(h': TestHelper) =>
+    h = h'
+    h.complete_action("client handler create called")
+
+  fun ref apply(payload: Payload val): Any => 
+    n_received = n_received + 1
+    h.complete_action("client handler apply called " + n_received.string())
+
+  fun ref chunk(data: ByteSeq val) => 
+    h.log("_HTTPConnTestHandler.chunk called")
+
+class val _HTTPConnTestHandlerFactory is HandlerFactory
+  let h: TestHelper
+
+  new val create(h': TestHelper) =>
+    h = h'
+
+  fun apply(session: HTTPSession): HTTPHandler ref^ =>
+    h.dispose_when_done(session)
+    h.complete_action("client factory apply called")
+    _HTTPConnTestHandler(h)
+
+class iso _HTTPConnTest is UnitTest
+  fun name(): String => "net/http/_HTTPConnection._new_conn"
+  fun label(): String => "conn-fix"
+
+  fun ref apply(h: TestHelper) ? =>
+    // Set expectations.
+    h.expect_action("client factory apply called")
+    h.expect_action("client handler create called")
+    h.expect_action("client handler apply called 1")
+    h.expect_action("client handler apply called 2")
+    h.expect_action("server writing reponse 1")
+    h.expect_action("server writing reponse 2")
+    h.expect_action("server listening")
+    h.expect_action("server listen connected")
+    h.expect_action("server connection accepted")
+    h.expect_action("server connection closed")
+
+    let worker = object
+      var client: (HTTPClient iso | None) = None
+
+      be listening(service: String) =>
+        try
+          // Need two or more request to check if the fix worked.
+          let loops: USize = 2
+          // let service: String val = "12345"
+          h.log("received service: [" + service + "]")
+          let us = "http://localhost:" + service
+          h.log("URL: " + us)
+          let url = URL.build(us)?
+          h.log("url.string()=" + url.string())
+          let hf = _HTTPConnTestHandlerFactory(h)
+          client = recover iso HTTPClient(h.env.root as TCPConnectionAuth) end
+
+          for _ in Range(0, loops) do 
+            let payload: Payload iso = Payload.request("GET", url)
+            payload.set_length(0)
+            try
+              (client as HTTPClient iso)(consume payload, hf)?
+            end
+          end
+        else 
+          h.log("Error in worker.listening")
+          h.complete(false)
+        end // try
+
+      be dispose() =>
+        // try
```

Comment:
do you want to uncomment those lines?

Reply:
Will do.

Reply:
I have to take it back. This was something that I put in to fix the problem on windows. Because it is not needed I will remove it from the code. As you mentioned earlier only one fix per pull request.

Reply:
This code is not necessary. I put it in hoping it would fix the problem on Windows. 

I will take it out.